src/main/steps/c100-rebuild/respondent-details/address/manual/ManualAddressPostController.ts

```diff
@@ -0,0 +1,46 @@
+import autobind from 'autobind-decorator';
+import { Response } from 'express';
+
+import { C100RebuildPartyDetails, YesOrNo } from '../../../../../app/case/definition';
+import { AppRequest } from '../../../../../app/controller/AppRequest';
+import { AnyObject, PostController } from '../../../../../app/controller/PostController';
+import { Form, FormFields, FormFieldsFn } from '../../../../../app/form/Form';
+import { AnyType } from '../../../../../app/form/validation';
+
+import { getUpdatedForm } from './content';
+
+@autobind
+export default class ManualAddressPostController extends PostController<AnyObject> {
+  constructor(protected readonly fields: FormFields | FormFieldsFn) {
+    super(fields);
+  }
+
+  public async post(req: AppRequest<AnyObject>, res: Response): Promise<void> {
+    const { respondentId } = req?.params;
+
+    const form = new Form(getUpdatedForm().fields as FormFields);
+    const { saveAndSignOut, saveBeforeSessionTimeout, _csrf, ...formData } = form.getParsedBody(req.body);
+
+    const respondentId1: AnyType | undefined = respondentId;
+
+    const respondent = this.getRespondentDetails(
+      //eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
+      req.session.userCase?.resp_Respondents!,
+      respondentId1
+    ) as C100RebuildPartyDetails;
+    respondent!.address!.PostCode = req.body['addressPostcode'] as string;
```

Comment:
We can use spread operator here I think to reduce these manual assignment.
@vivek-sekhar Can you please add your input ?